src/NSubstitute.Analyzers.Shared/Resources.resx

```diff
@@ -375,6 +375,19 @@
     <value>Argument matcher used with a non-virtual member of a class.</value>
         <comment>The title of the diagnostic.</comment>
   </data>
+  
+      <data name="WithAnyArgsArgumentMatcherUsageDescription" xml:space="preserve">
+    <value>Argument matcher used with WithAnyArgs method.</value>
```

Comment:
@dtchepak do you have any better alternative for warning title/description. IMO this one is not good enough

Reply:
Only thing I can think of is providing more information on the impact of this. Maybe something like:

```
Argument matcher used with WithAnyArgs. This matcher may not be executed.
```

?

Reply:
Looks better. Changed. 
One more thing though. `WithAnyArgs` naming is not 100% accurate, as same rules apply to `ReturnsForAnyArgs`, `WhenForAnyArgs` etc. Should we come up with some more generic naming or we are fine with `WithAnyArgs`?